NineChronicles.Headless.Tests/GraphTypes/StateQueryTest.cs

```diff
@@ -0,0 +1,222 @@
+                            var fig = new FungibleItemGarage(material, 10);
+                            arr[i] = fig.Serialize();
+                        }
+
+                        return arr;
+                    },
+                    (_, currency) => currency.Ticker switch
+                    {
+                        "NCG" => new FungibleAssetValue(currency, 99, 99),
+                        "CRYSTAL" or "GARAGE" => new FungibleAssetValue(
+                            currency,
+                            99,
+                            123456789012345678),
+                        _ => new FungibleAssetValue(currency, 99, 0),
+                    },
+                    0L));
+            Assert.Null(queryResult.Errors);
+            var data = (Dictionary<string, object>)((ExecutionNode)queryResult.Data!).ToValue()!;
+            var garages = (Dictionary<string, object>)data["garages"];
+            Assert.Equal(agentAddr.ToString(), garages["agentAddr"]);
+            Assert.Equal(Addresses.GetGarageBalanceAddress(agentAddr).ToString(), garages["garageBalancesAddr"]);
+            if (currencyEnums is not null)
+            {
+                var garageBalances = ((object[])garages["garageBalances"]).OfType<Dictionary<string, object>>();
+                Assert.Equal(currencyEnums.Count(), garageBalances.Count());
+                foreach (var (currencyEnum, garageBalance) in currencyEnums.Zip(garageBalances))
+                {
+                    Assert.Equal(
+                        currencyEnum.ToString(),
+                        ((Dictionary<string, object>)garageBalance["currency"])["ticker"]);
+                }
+            }
+            else if (currencyTickers is not null)
+            {
+                var garageBalances = ((object[])garages["garageBalances"]).OfType<Dictionary<string, object>>();
+                Assert.Equal(currencyTickers.Count(), garageBalances.Count());
+                foreach (var (currencyTicker, garageBalance) in currencyTickers.Zip(garageBalances))
+                {
+                    Assert.Equal(
+                        currencyTicker,
+                        ((Dictionary<string, object>)garageBalance["currency"])["ticker"]);
+                }
+            }
+
+            if (fungibleItemIds is not null)
+            {
+                var fungibleItemGarages =
+                    ((object[])garages["fungibleItemGarages"]).OfType<Dictionary<string, object>>();
+                Assert.Equal(fungibleItemIds.Count(), fungibleItemGarages.Count());
+                foreach (var (fungibleItemId, fungibleItemGarage) in fungibleItemIds.Zip(fungibleItemGarages))
+                {
+                    var actual = ((Dictionary<string, object>)fungibleItemGarage["item"])["fungibleItemId"];
+                    Assert.Equal(fungibleItemId, actual);
+                }
+            }
```

Comment:
I think we missed one case that the garage state is actually null.
I can see the query result in PR description but it's not covered here.